pkg/collector/horizontalpodautoscaler.go

```diff
@@ -35,16 +35,16 @@ func HorizontalPodAutoscaler(cfg config.Config, logger logr.Logger, otelcol v1al
 
 	return autoscalingv1.HorizontalPodAutoscaler{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:        naming.Collector(otelcol),
+			Name:        otelcol.Name,
 			Namespace:   otelcol.Namespace,
 			Labels:      labels,
 			Annotations: annotations,
 		},
 		Spec: autoscalingv1.HorizontalPodAutoscalerSpec{
 			ScaleTargetRef: autoscalingv1.CrossVersionObjectReference{
-				APIVersion: "apps/v1",
-				Kind:       "Deployment",
-				Name:       naming.Collector(otelcol),
+				APIVersion: "opentelemetry.io/v1alpha1",
+				Kind:       "OpenTelemetryCollector",
+				Name:       otelcol.Name,
 			},
 			MinReplicas:                    otelcol.Spec.Replicas,
```

Comment:
Another suggestion would be to remove the `maxReplicas` from the `OpenTelemetryCollectorSpec` and use something like:
```golang
type OpenTelemetryCollectorSpec struct {
	// [...]

	// Autoscaler specifies the pod autoscaling configuration to use
	// for the OpenTelemetryCollector workload.
	//
	// +optional
	Autoscaler *AutoscalerSpec `json:"autoscaler,omitempty"`

	// [...]
}

// AutoscalerSpec defines the OpenTelemetryCollector's pod autoscaling specification.
type AutoscalerSpec struct {
	// MinReplicas is the lower limit for the number of replicas to which the autoscaler
	// can scale down.  It defaults to 1 pod. MinReplicas is allowed to be 0 if the
	// alpha feature gate HPAScaleToZero is enabled and at least one Object or External
	// metric is configured. Scaling is active as long as at least one metric value is
	// available.
	//
	// +optional
	// +kubebuilder:validation:Minimum=0
	MinReplicas *int32 `json:"minReplicas,omitempty"`

	// MaxReplicas is the upper limit for the number of replicas to which the autoscaler can scale up.
	// It cannot be less than MinReplicas.
	// If no set, defaults to 100 replicas over MinReplicas.
	//
	// +optional
	// +kubebuilder:validation:Minimum=0
	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
}
```

This will allow you to add `Metrics` and `Behavior` once the version `v2` of the HPA will be GA.